File: solutions/_108_convert_sorted_array_to_binary_search_tree/Solution.java

package _108_convert_sorted_array_to_binary_search_tree;

import java.util.Arrays;

class Solution {
    public static TreeNode sortedArrayToBST(int[] nums) {
        if (nums.length == 0) {
            return null;
        }

        int mid = nums.length / 2;
        TreeNode node = new TreeNode(nums[mid]);

        node.left = sortedArrayToBST(Arrays.copyOfRange(nums, 0, mid));
        node.right = sortedArrayToBST(Arrays.copyOfRange(nums, mid + 1, nums.length));

        return node;
    }
}

/*
 * https://leetcode.com/problems/convert-sorted-array-to-binary-search-tree/
 */

File: solutions/_108_convert_sorted_array_to_binary_search_tree/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
var sortedArrayToBST = function (nums) {
  if (!nums.length) return null;

  const mid = Math.floor(nums.length / 2);
  const node = new TreeNode(nums[mid]);

  node.left = sortedArrayToBST(nums.slice(0, mid));
  node.right = sortedArrayToBST(nums.slice(mid + 1));

  return node;
};

module.exports = { sortedArrayToBST, TreeNode };

/*
https://leetcode.com/problems/convert-sorted-array-to-binary-search-tree/
*/
